### ISSUE TITLE
PhP warning - Cannot load Zend OPcache - it was already loaded

### ISSUE THREAD

**Issue by Iolaum** [nextcloud/nextcloudpi#416]
### PROBLEMS

I am not sure if this is a problem with ncp, php/raspbian or not a problem at all. Simply running `php -v` gives an error:

```
$ php -v
Cannot load Zend OPcache - it was already loaded
PHP 7.0.27-0+deb9u1 (cli) (built: Jan  5 2018 13:51:52) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.27-0+deb9u1, Copyright (c) 1999-2017, by Zend Technologies
```

Looking into some other log files as well, the error gets thrown at sys log every 30 min:

```
root@nextcloudpi:/var/log# cat syslog | grep OPcache
Feb 15 06:39:09 nextcloudpi sessionclean[10872]: Cannot load Zend OPcache - it was already loaded
Feb 15 06:39:09 nextcloudpi sessionclean[10872]: Cannot load Zend OPcache - it was already loaded
Feb 15 07:09:09 nextcloudpi sessionclean[11044]: Cannot load Zend OPcache - it was already loaded
Feb 15 07:09:09 nextcloudpi sessionclean[11044]: Cannot load Zend OPcache - it was already loaded
Feb 15 07:39:09 nextcloudpi sessionclean[11204]: Cannot load Zend OPcache - it was already loaded
Feb 15 07:39:10 nextcloudpi sessionclean[11204]: Cannot load Zend OPcache - it was already loaded
Feb 15 08:09:09 nextcloudpi sessionclean[11315]: Cannot load Zend OPcache - it was already loaded
...
```

**Paste the output of `sudo ncp-report`:**

<details>
<summary>NextCloudPi diagnostics</summary>

```

Cannot load Zend OPcache - it was already loaded
NextCloudPi version  v0.46.10
NextCloudPi image    NextCloudPi_12-04-17
distribution         Raspbian GNU/Linux 9 \n \l
automount            yes
USB devices          sda 
datadir              /media/USBdrive/ncdata
data in SD           no
data filesystem      ext2/ext3
data disk usage      182G/917G
rootfs usage         1.7G/3.6G
swapfile             /media/wd1/swap
Nextcloud check      ok
Nextcloud version    12.0.4.3
HTTPD service        up
PHP service          up
MariaDB service      up
Redis service        up
Postfix service      up
internet check       ok
port check 80        open
port check 443       open
IP                   192.168.1.20
gateway              192.168.1.1
interface            eth0
certificates         $HOSTNAME
certs due            25 days
NAT loopback         no
uptime               2days

```
</details>

<details>
<summary>Nextcloud configuration</summary>

```

Cannot load Zend OPcache - it was already loaded
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": {
            "0": "$HOSTNAME:443",
            "1": "192.168.1.20",
            "4": "$HOSTNAME",
            "2": "$HOSTNAME"
        },
        "datadirectory": "\/media\/USBdrive\/ncdata",
        "overwrite.cli.url": "https:\/\/$HOSTNAME",
        "dbtype": "mysql",
        "version": "12.0.4.3",
        "dbname": "nextcloud",
        "dbhost": "localhost",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "ocs86rkb5syt",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "\/var\/run\/redis\/redis.sock",
            "port": 0,
            "timeout": 0,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "mail_smtpmode": "php",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "admin",
        "mail_domain": "$HOSTNAME",
        "overwriteprotocol": "https",
        "maintenance": false,
        "loglevel": "2",
        "logfile": "\/media\/wd1\/ncdata\/nextcloud.log",
        "log_type": "file"
    }
}

```
</details>

<details>
<summary>HTTPd logs</summary>

```

[Thu Feb 15 06:25:05.002191 2018] [mpm_event:notice] [pid 859:tid 1992183808] AH00489: Apache/2.4.25 (Raspbian) OpenSSL/1.0.2l   configured -- resuming normal operations
[Thu Feb 15 06:25:05.002289 2018] [core:notice] [pid 859:tid 1992183808] AH00094: Command line: '/usr/sbin/apache2'

```
</details>

<details>
<summary>Database logs</summary>

```

tail: cannot open '/var/log/mysql/*.log' for reading: No such file or directory

```
</details>

<details>
<summary>Nextcloud logs</summary>

```

{"reqId":"WoCkVX8AAQEAAAG4u6UAAA4B","level":2,"time":"2018-02-11T20:15:18+00:00","remoteAddr":"137.226.113.20","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"137.226.113.20\" tried to access using \"79.75.141.71\" as host.","userAgent":"Mozilla\/5.0 (X11; Ubuntu; Linux x86_64; rv:50.0) Gecko\/20100101 Firefox\/50.0","version":"12.0.4.3"}
{"reqId":"WoCkVn8AAQEAAAG4u60AAA4J","level":2,"time":"2018-02-11T20:15:19+00:00","remoteAddr":"137.226.113.20","user":"--","app":"core","method":"GET","url":"\/index.php\/js\/core\/merged-template-prepend.js?v=2793255a2c31f4b4c2eab906af15ce25-1","message":"Trusted domain error. \"137.226.113.20\" tried to access using \"79.75.141.71\" as host.","userAgent":"Mozilla\/5.0 (X11; Ubuntu; Linux x86_64; rv:50.0) Gecko\/20100101 Firefox\/50.0","version":"12.0.4.3"}
{"reqId":"WoDQD38AAQEAAAHFrL4AAABS","level":2,"time":"2018-02-11T23:21:51+00:00","remoteAddr":"107.170.227.152","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"107.170.227.152\" tried to access using \"79.75.141.71\" as host.","userAgent":"Mozilla\/5.0 zgrab\/0.x","version":"12.0.4.3"}
{"reqId":"WoF0O38AAQEAAB-8900AAAAA","level":3,"time":"2018-02-12T11:02:19+00:00","remoteAddr":"210.16.189.29","user":"--","app":"PHP","method":"GET","url":"\/admin\/config.php","message":"Exception: The requested uri(\/admin\/config.php) cannot be processed by the script '\/core\/templates\/404.php') at \/var\/www\/nextcloud\/lib\/private\/AppFramework\/Http\/Request.php#729","userAgent":"curl\/7.19.7 (i386-redhat-linux-gnu) libcurl\/7.19.7 NSS\/3.27.1 zlib\/1.2.3 libidn\/1.18 libssh2\/1.4.2","version":"12.0.4.3"}
{"reqId":"WoHJ6H8AAQEAAB-893IAAAAO","level":2,"time":"2018-02-12T17:07:53+00:00","remoteAddr":"164.52.6.150","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"164.52.6.150\" tried to access using \"79.75.151.21:443\" as host.","userAgent":"Mozilla\/5.0 (Windows NT 6.1; WOW64; rv:52.0) Gecko\/20100101 Firefox\/52.0","version":"12.0.4.3"}
{"reqId":"WoHjsn8AAQEAAB-8948AAAAS","level":2,"time":"2018-02-12T18:57:55+00:00","remoteAddr":"209.126.136.7","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"209.126.136.7\" tried to access using \"79.75.151.21\" as host.","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/57.0.2987.133 Safari\/537.36","version":"12.0.4.3"}
{"reqId":"WoH69X8AAQEAACAK1dIAAABE","level":2,"time":"2018-02-12T20:37:09+00:00","remoteAddr":"139.162.78.135","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"139.162.78.135\" tried to access using \"79.75.151.21\" as host.","userAgent":"Mozilla\/5.0 (Windows NT 10.0; WOW64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/45.0.2454.101 Safari\/537.36","version":"12.0.4.3"}
{"reqId":"WoJC9H8AAQEAACAK9L8AAABL","level":2,"time":"2018-02-13T01:44:20+00:00","remoteAddr":"34.214.38.179","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"34.214.38.179\" tried to access using \"92.4.153.185:443\" as host.","userAgent":"Mozilla\/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/50.0.2661.102 Safari\/537.36","version":"12.0.4.3"}
{"reqId":"WoLhKn8AAQEAAD5A4SIAAABL","level":2,"time":"2018-02-13T12:59:22+00:00","remoteAddr":"139.162.78.135","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"139.162.78.135\" tried to access using \"92.4.153.185\" as host.","userAgent":"Mozilla\/5.0 (Windows NT 10.0; WOW64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/45.0.2454.101 Safari\/537.36","version":"12.0.4.3"}
{"reqId":"WoLrIn8AAQEAAD4wuFkAAAAH","level":2,"time":"2018-02-13T13:41:54+00:00","remoteAddr":"184.105.139.69","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"184.105.139.69\" tried to access using \"92.4.153.185\" as host.","userAgent":"--","version":"12.0.4.3"}
{"reqId":"WoMy238AAQEAAD4wxZkAAAAS","level":2,"time":"2018-02-13T18:47:55+00:00","remoteAddr":"107.170.227.37","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"107.170.227.37\" tried to access using \"79.75.128.116\" as host.","userAgent":"Mozilla\/5.0 zgrab\/0.x","version":"12.0.4.3"}
{"reqId":"WoOZQX8AAQEAAANxI7kAAAAT","level":2,"time":"2018-02-14T02:04:49+00:00","remoteAddr":"89.248.160.199","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"89.248.160.199\" tried to access using \"79.75.128.116\" as host.","userAgent":"libwww-perl\/6.31","version":"12.0.4.3"}
{"reqId":"WoQxAn8AAQEAAA2rWeEAAABQ","level":2,"time":"2018-02-14T12:52:19+00:00","remoteAddr":"184.105.139.69","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"184.105.139.69\" tried to access using \"79.75.128.116\" as host.","userAgent":"--","version":"12.0.4.3"}
{"reqId":"WoR7bH8AAQEAAA2rWg8AAABV","level":2,"time":"2018-02-14T18:09:48+00:00","remoteAddr":"60.191.38.77","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"60.191.38.77\" tried to access using \"79.75.128.116:443\" as host.","userAgent":"Mozilla\/5.0 (Macintosh; Intel Mac OS X 10.11; rv:47.0) Gecko\/20100101 Firefox\/47.0","version":"12.0.4.3"}
{"reqId":"WoSXW38AAQEAAA2rWicAAABE","level":3,"time":"2018-02-14T20:08:59+00:00","remoteAddr":"163.172.192.121","user":"--","app":"PHP","method":"GET","url":"\/a2billing\/common\/javascript\/misc.js","message":"Exception: The requested uri(\/a2billing\/common\/javascript\/misc.js) cannot be processed by the script '\/core\/templates\/404.php') at \/var\/www\/nextcloud\/lib\/private\/AppFramework\/Http\/Request.php#729","userAgent":"python-requests\/2.18.4","version":"12.0.4.3"}
{"reqId":"WoT5@38AAQEAAA2rWzoAAABJ","level":2,"time":"2018-02-15T03:09:47+00:00","remoteAddr":"139.162.78.135","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"139.162.78.135\" tried to access using \"79.75.128.116\" as host.","userAgent":"Mozilla\/5.0 (Windows NT 10.0; WOW64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/45.0.2454.101 Safari\/537.36","version":"12.0.4.3"}
{"reqId":"WoUEsX8AAQEAAA2rWzwAAABM","level":3,"time":"2018-02-15T03:55:29+00:00","remoteAddr":"163.172.192.121","user":"--","app":"PHP","method":"GET","url":"\/a2billing\/common\/javascript\/misc.js","message":"Exception: The requested uri(\/a2billing\/common\/javascript\/misc.js) cannot be processed by the script '\/core\/templates\/404.php') at \/var\/www\/nextcloud\/lib\/private\/AppFramework\/Http\/Request.php#729","userAgent":"python-requests\/2.18.4","version":"12.0.4.3"}
{"reqId":"WoUVEH8AAQEAAA2i0dsAAAAL","level":2,"time":"2018-02-15T05:05:20+00:00","remoteAddr":"80.82.77.139","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"80.82.77.139\" tried to access using \"79.75.128.116\" as host.","userAgent":"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/34.0.1847.137 Safari\/537.36","version":"12.0.4.3"}
{"reqId":"WoV-C38AAQEAACmpy1YAAAAU","level":2,"time":"2018-02-15T12:37:31+00:00","remoteAddr":"216.218.206.66","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"216.218.206.66\" tried to access using \"79.75.156.41\" as host.","userAgent":"--","version":"12.0.4.3"}
{"reqId":"WoX18X8AAQEAACmpzJsAAAAH","level":2,"time":"2018-02-15T21:04:50+00:00","remoteAddr":"60.191.38.77","user":"--","app":"core","method":"GET","url":"\/","message":"Trusted domain error. \"60.191.38.77\" tried to access using \"79.75.156.41:443\" as host.","userAgent":"Mozilla\/5.0 (Macintosh; Intel Mac OS X 10.11; rv:47.0) Gecko\/20100101 Firefox\/47.0","version":"12.0.4.3"}

```
</details>

**Comment by nachoparker** [comment]
That log is not the mysql log, but the apache log. See this

https://github.com/nextcloud/nextcloudpi/blob/master/etc/nextcloudpi-config.d/nc-nextcloud.sh#L190

You are right, my mysql log is also empty... let's see if we can find it. Are you using ramlogs? does it come back after doing `systemctl restart mysqld`?

**Comment by nachoparker** [comment]
I think I found it. Some update duplicated the symlink for the opcache configuration

```
# ls -l /etc/php/7.0/fpm/conf.d/ | grep opc
lrwxrwxrwx 1 root root 39 Dec  4 13:29 10-opcache.ini -> /etc/php/7.0/mods-available/opcache.ini
lrwxrwxrwx 1 root root 39 Feb  7 10:47 20-opcache.ini -> /etc/php/7.0/mods-available/opcache.ini
```

Can you replicate this finding?

It is just a harmless warning.

**Comment by Iolaum** [comment]
Yes I can replicate that:

```
$ pwd
/etc/php/7.0/fpm/conf.d
$ ll | grep opcache
lrwxrwxrwx 1 root root   39 Dec  4 13:29 10-opcache.ini -> /etc/php/7.0/mods-available/opcache.ini
lrwxrwxrwx 1 root root   39 Jan 14 22:49 20-opcache.ini -> /etc/php/7.0/mods-available/opcache.ini
```
Is the solution to just delete the second symlink?

**Comment by nachoparker** [comment]
yes, I fixed it in the update, please reopen if it still happens after `v0.46.16`

**Comment by Iolaum** [comment]
It looks like this bug reappeared as we moved to php7.2:

```
$ echo "$( ls -l /etc/php/7.2/fpm/conf.d/*-opcache.ini |  wc -l )"
2
$ echo "$( ls -l /etc/php/7.2/cli/conf.d/*-opcache.ini |  wc -l )"
2

```

**Comment by ovpc** [comment]
Strange, it only happens on my rpi3B+ , not on my Debian9 install

**Comment by Iolaum** [comment]
I think this bug creeped in after nacho enabled the buster sources, not php7.2 but I am not 100% sure

**Comment by ovpc** [comment]
2 simlinks in /etc/php/7.2/cli/conf.d/  (10- and 20-) both pointing to    /etc/php/7.2/mods-available/opcache.ini

**Comment by eogut** [comment]
> Back again ;-( 2 simlinks in /etc/php/7.2/cli/conf.d/ (10- and 20-) both pointing to /etc/php/7.2/mods-available/opcache.ini

I have that error, too!

**Comment by nachoparker** [comment]
should be fixed now again. Note that this is a harmless warning, that appears with updates. Not much we can do about it.

**Comment by saidogoid** [comment]
Hi,


In amazon Ami2 solved to rename opcache.ini to 10-opcache.ini.
Go to /etc/php.d/ and rename it. 
See if you have here a duplicated opcache too.

Cheers

**Comment by amilopowers** [comment]
This still happens in v1.11.2 on ODROID XU4.

**Comment by miri64** [comment]
Had the same problem in v1.17.1 (after an update on Sep 30th) and removing one of the symlink solved it

**Comment by Iolaum** [comment]
@miri64 which symlink? I noticed in my ncp gives that warning again.

**Comment by miri64** [comment]
See https://github.com/nextcloud/nextcloudpi/issues/416#issuecomment-366183822. I just deleted the older of the two to silence the warnings.

**Comment by Iolaum** [comment]
Hm, so it's the same issue again. Thought that the upgrade to buster would have solved that.

**Comment by ovpc** [comment]
Returned again
Fix is
> sudo rm /etc/php/7.3/cli/conf.d/20-opcache.ini
> sudo rm /etc/php/7.3/fpm/conf.d/20-opcache.ini

**Comment by ferderm** [comment]
Hi there,

with the updates to `php7.3-opcache armhf 7.3.27-1~deb10u1` / `php7.3 all 7.3.27-1~deb10u1` the issue re-appaered again. 

> sudo rm /etc/php/7.3/cli/conf.d/20-opcache.ini
> sudo rm /etc/php/7.3/fpm/conf.d/20-opcache.ini
 
did it.

**Comment by ovpc** [comment]
yep, same here.

**Comment by Iolaum** [comment]
It appears a PR similar to [this one](https://github.com/nextcloud/nextcloudpi/pull/1031/files) is needed for this re-appearing in php7.3 series. If no-one is up for it I 'll try to find some time and pick it up.

**Comment by FadeFx** [comment]
seems to me the problem is back, i just restored my nextcloud onto a fresh installation of ncp after apt-get upgrade (raspberrypi) and see this during most operations.
ncp version 1.48.1
php 7.4

**Comment by thatgeek** [comment]
for me, this came up with a webuzo update to php. the extra.ini file at /usr/local/apps/php74/etc/php.d/extra.ini had two lines duplicated:
```
zend_extension=ioncube_loader_lin_7.4.so
zend_extension=opcache.so
```

removing the duplicated lines solved the warnings.

**Comment by theCalcaholic** [comment]
@thatgeek What's a webuzo update? 🤔

**Comment by deathontheshore** [comment]
Seems to have reappeared:
`lrwxrwxrwx 1 root root  39 Jan 22 02:31 10-opcache.ini -> /etc/php/8.1/mods-available/opcache.ini
lrwxrwxrwx 1 root root  39 Feb  6 11:46 20-opcache.ini -> /etc/php/8.1/mods-available/opcache.ini`

**Comment by token0** [comment]
Manual fix.
Find all occurrences:
```
# for f in $(find /etc/php/ 2>/dev/null -name "*opc*") ; do ls -laFh $f;done
lrwxrwxrwx 1 root root 39 Dec 29  2022 /etc/php/7.3/cli/conf.d/20-opcache.ini -> /etc/php/7.3/mods-available/opcache.ini
lrwxrwxrwx 1 root root 39 Jul  6  2020 /etc/php/7.3/cli/conf.d/10-opcache.ini -> /etc/php/7.3/mods-available/opcache.ini
lrwxrwxrwx 1 root root 39 Dec 29  2022 /etc/php/7.3/cgi/conf.d/20-opcache.ini -> /etc/php/7.3/mods-available/opcache.ini
lrwxrwxrwx 1 root root 39 Jul  6  2020 /etc/php/7.3/cgi/conf.d/10-opcache.ini -> /etc/php/7.3/mods-available/opcache.ini
lrwxrwxrwx 1 root root 39 Dec 29  2022 /etc/php/7.3/fpm/conf.d/20-opcache.ini -> /etc/php/7.3/mods-available/opcache.ini
```
And then remove oldest duplicates. In my case those that was dated 2020.
```
rm /etc/php/7.3/cli/conf.d/10-opcache.ini
rm /etc/php/7.3/cgi/conf.d/10-opcache.ini
```

**Comment by miri64** [comment]
This showed up again after updating the PHP packages to 8.1.29-1+0~20240606.65+debian12~1.gbp2ddf4e last Saturday (2024-06-08). I am on NCP v1.52.2:

```console
$ ls -l /etc/php/8.1/*/conf.d/*opcache*.ini
lrwxrwxrwx 1 root root 39 May  2 03:15 /etc/php/8.1/cli/conf.d/10-opcache.ini -> /etc/php/8.1/mods-available/opcache.ini
lrwxrwxrwx 1 root root 39 Jun  8 20:44 /etc/php/8.1/cli/conf.d/20-opcache.ini -> /etc/php/8.1/mods-available/opcache.ini
lrwxrwxrwx 1 root root 39 May  2 03:15 /etc/php/8.1/fpm/conf.d/10-opcache.ini -> /etc/php/8.1/mods-available/opcache.ini
lrwxrwxrwx 1 root root 39 Jun  8 20:44 /etc/php/8.1/fpm/conf.d/20-opcache.ini -> /etc/php/8.1/mods-available/opcache.ini
```

**Comment by Matthionine** [comment]
This appears again with the update to php 8.3 and is resolved as before on ncp 1.58:
```
$ ls -l /etc/php/8.*/*/conf.d/*opcache*.ini
lrwxrwxrwx 1 root root 39 Sep 10  2024 /etc/php/8.3/cli/conf.d/10-opcache.ini -> /etc/php/8.3/mods-available/opcache.ini
lrwxrwxrwx 1 root root 39 Jul 17 04:33 /etc/php/8.3/cli/conf.d/20-opcache.ini -> /etc/php/8.3/mods-available/opcache.ini
lrwxrwxrwx 1 root root 39 Sep 10  2024 /etc/php/8.3/fpm/conf.d/10-opcache.ini -> /etc/php/8.3/mods-available/opcache.ini
lrwxrwxrwx 1 root root 39 Jul 17 04:33 /etc/php/8.3/fpm/conf.d/20-opcache.ini -> /etc/php/8.3/mods-available/opcache.ini
```

and can be fixed by deleting the additional .ini-file:
```
$ sudo rm /etc/php/8.1/*/conf.d/20-opcache.ini
```
